integration-tests/startup/index.js

```diff
@@ -16,6 +16,7 @@ require('dd-trace').init(options)
 if (process.env.STEALTHY_REQUIRE === 'true') {
   // emulate stealthy-require
   for (const key in require.cache) {
+    if (key.endsWith('dd-trace/packages/datadog-core/index.js')) continue
```

Comment:
@bengl You sure we should merge that ? I don't even know why it works

Reply:
My guess on what is happening here: 
This stealthy require integration test fails without this line because it expects outbounds requests from the tracer itself to not be traced.
But by deleting the require cache of the datadog-core AsyncLocalStorage instance (as shown above) it creates two different instances of AsyncLocalStorage and breaks the code that prevents the tracer from tracing itself.
So by adding this line, I prevent this ALS instance mismatch and the integration test passes.

Happy to show you on zoom if the explanation is too obscure.
cc: @tlhunter

Reply:
My intuition is we'll need to find another solution to this.

Reply:
Doesn't that mean we'd have the same issue when a customer is actually using `stealthy-require`?

Reply:
well if stealthy-require messes with our datadog-core storage instance, then yeah I guess the tracer could be acting weird, but it's not a problem with this PR, it's a problem that already exists in the tracer.

Reply:
Here is the PR to separate the issue